### PR TITLE
KnnVectorQueryBuilder  support for allowUnmappedFields

### PR DESCRIPTION
**Pull request by ghostspiders**
KnnVectorQueryBuilder does not respect the index.query.parse.allow_unmapped_fields index setting. update KnnVectorQueryBuilder  support for allowUnmappedFields.
https://github.com/elastic/elasticsearch/issues/106846

### CODE REVIEW

**Comment by benwtrent** [comment]
I think the change looks good. Gonna get CI to test. Thank you @ghostspiders !

**Comment by benwtrent** [comment]
build kite test this please

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
@ghostspiders 

There are multiple failures around the yaml tests:

> ./gradlew ':qa:mixed-cluster:v8.13.3#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/45_knn_search_byte/Test nonexistent field}" -Dtests.seed=A52D9437EA192750 -Dtests.bwc=true -Dtests.locale=it-IT -Dtests.timezone=Africa/Mbabane -Druntime.java=21

is one example.

I am thinking that those tests need to adjusted with a version skip to ensure that they are only ran on older versions where we would fail on a non-existent field directly.

**Comment by ghostspiders** [comment]
@benwtrent  I passed this yaml tests
`./gradlew  ':qa:mixed-cluster:v8.9.2#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/45_knn_search_byte/Test nonexistent field}"  -Dtests.bwc=true  -Druntime.java=21
./gradlew  ':qa:mixed-cluster:v8.9.2#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/40_knn_search/Test nonexistent field}"  -Dtests.bwc=true  -Druntime.java=21`

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
buildkite test this please

**Comment by benwtrent** [comment]
buildkite test this please

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml [hunk 1]
```diff
@@ -287,6 +287,9 @@ setup:
   - requires:
       cluster_features: "gte_v8.4.0"
       reason: 'kNN added to search endpoint in 8.4'
+  - skip:
+      cluster_features: "gte_v8.15.0"
+      reason: 'non-existent field handling improved in 8.15'
```
Comment (benwtrent) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml:292]:
```suggestion
      cluster_features: "gte_v8.16.0"
      reason: 'non-existent field handling improved in 8.16'
```

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml [hunk 2]
```diff
@@ -298,9 +301,28 @@ setup:
...
             num_candidates: 3
+
   - match: { error.root_cause.0.type: "query_shard_exception" }
   - match: { error.root_cause.0.reason: "failed to create query: field [nonexistent] does not exist in the mapping" }
 
+---
+"Test nonexistent field is match none":
+  - requires:
+      cluster_features: "gte_v8.15.0"
+      reason: 'non-existent field handling improved in 8.15'
```
Comment (benwtrent) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/40_knn_search.yml:312]:
```suggestion
      cluster_features: "gte_v8.16.0"
      reason: 'non-existent field handling improved in 8.16'
```

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/45_knn_search_byte.yml [hunk 3]
```diff
@@ -148,6 +148,9 @@ setup:
 
 ---
 "Test nonexistent field":
+  - skip:
+      cluster_features: 'gte_v8.15.0'
+      reason: 'non-existent field handling improved in 8.15'
```
Comment (benwtrent) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/45_knn_search_byte.yml:153]:
```suggestion
      cluster_features: 'gte_v8.16.0'
      reason: 'non-existent field handling improved in 8.16'
```

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/45_knn_search_byte.yml [hunk 4]
```diff
@@ -159,8 +162,26 @@ setup:
...
             k: 2
             num_candidates: 3
+
   - match: { error.root_cause.0.type: "query_shard_exception" }
   - match: { error.root_cause.0.reason: "failed to create query: field [nonexistent] does not exist in the mapping" }
+---
+"Test nonexistent field is match none":
+  - requires:
+      cluster_features: 'gte_v8.15.0'
+      reason: 'non-existent field handling improved in 8.15'
```
Comment (benwtrent) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/45_knn_search_byte.yml:172]:
```suggestion
      cluster_features: 'gte_v8.16.0'
      reason: 'non-existent field handling improved in 8.16'
```

**Review by benwtrent — COMMENTED** [review]
gotta update for 8.16

**Comment by benwtrent** [comment]
buildkite test this please

**Review by benwtrent — APPROVED** [review]

**Pull request merged**